test/NuGet.Core.Tests/NuGet.Protocol.Tests/HttpSource/HttpSourceTests.cs

```diff
@@ -449,9 +449,41 @@ public async Task HttpSource_ValidatesInvalidCachedContent()
             }
         }
 
+        [Fact]
+        public async Task HttpSource_ThrottlesRequestsFromSettings()
+        {
+            // Arrange
+            using (var td = TestDirectory.Create())
+            {
+                var tc = new TestContext(td, SemaphoreSlimThrottle.CreateSemaphoreThrottle(initialCount: 4));
+
+                tc.MessageHandler.WaitTimeInMs = 100;
+                var tasks = new List<Task>();
+
+                // Act
+
+                for (var i = 0; i < 18; i++)
+                {
+                    tasks.Add(Task.Run(() =>
+                    {
+                        return tc.HttpSource.ProcessStreamAsync(
+                            new HttpSourceRequest(tc.Url, tc.Logger),
+                            stream => Task.FromResult(true),
+                            tc.Logger,
+                            token: CancellationToken.None);
+                    }));
+                }
+
+                await Task.WhenAll(tasks);
+
+                // Assert
+                Assert.True(tc.MessageHandler.MaxConcurrencyRequest <= 4);
```

Comment:
I'm not following this test, by this line, all tasks have been awaited so wouldn't it always be 0?

Ideally you should check for `MaxConcurrencyRequest` before `await Task.WhenAll(tasks);`

Reply:
discussed offline, we never decrement `MaxConcurrencyRequest ` so once it reaches 4 it stays as is. So it should be good.